src/core/VarReplay.jl

```diff
@@ -102,14 +103,15 @@ mutable struct VarInfo
     end
 end
 
-function Turing.runmodel!(model::Model, vi::VarInfo, spl::Union{Nothing,Sampler})
+function Turing.runmodel!(model::Model, vi::VarInfo, spl::AbstractSampler)
```

Comment:
Would it make sense to use a generated function here?

Reply:
Hmm... I think it's not a big impact: a generated function only change the `isa` type-check to a multi-dispatch. Anyway, I changed it to a generated function.